feat: alert button for updates, remind every 12 hours

cdrage opened the pull request:
### What does this PR do?

* Adds an "update" button on the bottom right that spins when an update
  is available
* Podman Desktop will check every 12 hours that it's running for any new
  updates

### Screenshot/screencast of this PR

<!-- Please include a screenshot or a screencast explaining what is doing this PR -->
![Screenshot 2023-02-23 at 11 51 32 AM](https://user-images.githubusercontent.com/6422176/220977124-75b7197a-3939-4942-9842-378374716420.png)
![Screenshot 2023-02-23 at 11 12 49 AM](https://user-images.githubusercontent.com/6422176/220977126-cdd8dd16-b655-42e1-aca6-939a2235bb77.png)



### What issues does this PR fix or reference?

<!-- Please include any related issue from Podman Desktop repository (or from another issue tracker).
-->

Closes https://github.com/containers/podman-desktop/issues/395
Closes https://github.com/containers/podman-desktop/issues/1418

### How to test this PR?

<!-- Please explain steps to reproduce -->

Follow the updated CONTRIBUTING.md guide

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

benoitf commented on packages/main/src/index.ts:
```diff
@@ -56,21 +92,13 @@ if (isWindows) {
- * Check new app version in production mode only
- */
-if (import.meta.env.PROD) {
-  app
```
it seems you removed the fact to enable this behavior only for production builds ?

cdrage replied:
still WIP, was removed for testing, latest PR pushed adds this back.

I'll ping once ready for review!

cdrage commented:
Ready for review! The review process may take longer than usual since if you want to test the "update" production functionality, you'll have to setup a minio server.

There is also a UI/UX change @benoitf that we may need feedback on. 

Since it's in the status-bar, I used `statusbar-item.ts` and applied styling by creating a boolean "notify" and detecting it within Svelte. This will come in hand in the future as we create more status bar items.

lstocchi commented:
Tried the first part. This is how it looks on Windows. LGTM so far

![notification_demo](https://user-images.githubusercontent.com/49404737/217576636-a07b8d51-c333-40a3-a0ae-6b14a2daf443.gif)

If i click on `update` nothing happen, is it right?
Other thing which is not related ot this PR is ... should we display the podman desktop version somewhere? I don't know which version i'm using otherwise.

Moving to the second part 👍

benoitf commented:
@lstocchi version should be visible in help/about menu

cdrage commented:
> 

Yup! To test the update part, the guide is in the updated CONTRIBUTING.md doc.

cdrage commented:
> Tried the first part. This is how it looks on Windows. LGTM so far
> 
> ![notification_demo](https://user-images.githubusercontent.com/49404737/217576636-a07b8d51-c333-40a3-a0ae-6b14a2daf443.gif) [ ![notification_demo](https://user-images.githubusercontent.com/49404737/217576636-a07b8d51-c333-40a3-a0ae-6b14a2daf443.gif) ](https://user-images.githubusercontent.com/49404737/217576636-a07b8d51-c333-40a3-a0ae-6b14a2daf443.gif) [ ](https://user-images.githubusercontent.com/49404737/217576636-a07b8d51-c333-40a3-a0ae-6b14a2daf443.gif)
> 
> If i click on `update` nothing happen, is it right? Other thing which is not related ot this PR is ... should we display the podman desktop version somewhere? I don't know which version i'm using otherwise. The only way is clicking on the notification button
> 
> Moving to the second part 👍

Yup! To test the update part, the guide is in the updated CONTRIBUTING.md doc which is a bit more involved to test.

lstocchi commented:
@cdrage i keep facing this error `NoCredentialProviders: no valid providers in chain. Deprecated.` any idea?

My current situation is this one. I created an ec2 instance, installed podman, run the command in the doc to run minio server, logged in into it, created a bucket with name `test-update`, created an access key, used these value as credential in the command found in the doc and run it... am i doing something wrong?

cdrage commented:
> @cdrage i keep facing this error `NoCredentialProviders: no valid providers in chain. Deprecated.` any idea?
> 
> My current situation is this one. I created an ec2 instance, installed podman, run the command in the doc to run minio server, logged in into it, created a bucket with name `test-update`, created an access key, used these value as credential in the command found in the doc and run it... am i doing something wrong?

Did you make sure to pass in the AWS credentials to your command line and update the external IP? 
```
AWS_ACCESS_KEY_ID=ID AWS_SECRET_ACCESS_KEY=SECRETKEY cross-env MODE=production npm run build && electron-builder build --publish always --config .electron-builder.config.cjs --config.publish.provider=s3 --config.publish.endpoint=http://<external-ip>:9000 --config.publish.bucket=test-update
```

cdrage commented:
Updated the PR with more documentation @lstocchi

slemeur commented:
@cdrage : Great work ! If the user clicks on "Later" what is happening? is the user able to find the information back somewhere? (I'd expect that this would be available in Podman Desktop application's menu)

The "update" icon looks like a "refresh" button, and may not be "self explicit" to all users. I can't see in the recording if there is a tooltip appearing? Should we add a label "an update is available" directly in the status bar? cc @mairin

cdrage commented:
> @cdrage : Great work ! If the user clicks on "Later" what is happening? is the user able to find the information back somewhere? (I'd expect that this would be available in Podman Desktop application's menu)
> 
> The "update" icon looks like a "refresh" button, and may not be "self explicit" to all users. I can't see in the recording if there is a tooltip appearing? Should we add a label "an update is available" directly in the status bar? cc @mairin

When it clicks on later, the update is dismissed / notification dismissed and won't appear until the next check (in 12 hours) so it doesn't annoy the user as much. The icon does reappear as well if the user closes and reopens Podman Desktop again.

UI/UX not finalized yet so would like @mairin 's input. However, we won't be able to do much UI changes except for icon / colour since we'll have to refactor the way that we output status bar items (currently there's not much styling we can do)

slemeur commented:
Thanks @cdrage .

> When it clicks on later, the update is dismissed / notification dismissed and won't appear until the next check (in 12 hours) so it doesn't annoy the user as much. The icon does reappear as well if the user closes and reopens Podman Desktop again.

Shouldn't we have the action available in Podman Desktop application's menu?

cdrage commented:
> Thanks @cdrage .
> 
> > When it clicks on later, the update is dismissed / notification dismissed and won't appear until the next check (in 12 hours) so it doesn't annoy the user as much. The icon does reappear as well if the user closes and reopens Podman Desktop again.
> 
> Shouldn't we have the action available in Podman Desktop application's menu?

Sure. Maybe don't entirely dismiss the icon but make it stop showing pulsing orange?

Thoughts @mairin

deboer-tim commented:
+1 to keep icon but stop pulsing.

cdrage commented:
> +1 to keep icon but stop pulsing.

Thanks! Pushed the update to disable the icon pulsing orange, but keep the icon there.

If there is no update available, then the icon will not appear at all.

lstocchi commented:
> Updated the PR with more documentation @lstocchi

Thanks @cdrage. Now i'm able to push everyting on my minio server but when i try to update it, the message says that the version 0.0.11 is the newest (but i have the 0.0.2 in minio) and it does nothing. I have tried on friday and today multiple times so  i  am missing something. Any idea? 🤔

cdrage commented:
Did you make sure that the latest.yml on the minio server says 0.11 is the
newest?

What does the console log say?

On Mon, Feb 13, 2023 at 06:18 Luca Stocchi ***@***.***> wrote:

> Updated the PR with more documentation @lstocchi
> <https://github.com/lstocchi>
>
> Thanks @cdrage <https://github.com/cdrage>. Now i'm able to push
> everyting on my minio server but when i try to update it, the message says
> that the version 0.0.11 is the newest (but i have the 0.0.2 in minio) and
> it does nothing. I have tried on friday and today multiple times so i am
> missing something. Any idea? 🤔
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/containers/podman-desktop/pull/1386#issuecomment-1427767690>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABQ75IDLBB6BTINGQBPZK7DWXIKBJANCNFSM6AAAAAAUTENVIE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
--
Charlie Drage
PGP - 4096R/C037D617
GPG / Encrypted Communication: charliedrage.com/contact

lstocchi commented:
> Did you make sure that the latest.yml on the minio server says 0.11 is the newest? What does the console log say?

I have this as latest.yaml

```
version: 0.0.2
files:
  - url: podman-desktop-0.0.2-setup.exe
    sha512: E6dDGpgywFdYwufFWX1yKGplV/CHoQe8BOLVW5gC2aprjK8EA+ucrml7s3AXbhiJi+qX4hhDZIugW7aCJRpJoA==
    size: 106385654
path: podman-desktop-0.0.2-setup.exe
sha512: E6dDGpgywFdYwufFWX1yKGplV/CHoQe8BOLVW5gC2aprjK8EA+ucrml7s3AXbhiJi+qX4hhDZIugW7aCJRpJoA==
releaseDate: '2023-02-10T14:06:55.086Z'
```

i download the 0.0.1, and i see 
![immagine](https://user-images.githubusercontent.com/49404737/218470569-24d8a0b4-af63-45b2-9593-846c9ac8cbc8.png)

The console doesn't say much 
![immagine](https://user-images.githubusercontent.com/49404737/218470875-42c8d6d2-ebae-4fbc-bfc9-bd371e7a5af3.png)

lstocchi commented:
The weird thing is that if i add 
```
publish: {
    provider: 's3',
    endpoint: 'http://ec2-****.compute-1.amazonaws.com:9090',
    bucket: 'test-update',    
    timeout: 10000,
  }
```

it still seems it's pointing to github ...

cdrage commented:
> The weird thing is that if i add
> 
> ```
> publish: {
>     provider: 's3',
>     endpoint: 'http://ec2-****.compute-1.amazonaws.com:9090',
>     bucket: 'test-update',    
>     timeout: 10000,
>   }
> ```
> 
> it still seems it's pointing to github ...

Yup, that's normal and part of the build process.. it's the reason why people complain about it here: https://github.com/electron-userland/electron-builder/issues/3053

lstocchi commented:
No it still doesn't work for me :( 
@jeffmaury when you have time can you try on your machine? I can also borrow you a minio server. I have the free trial in aws. Just ping me.

benoitf commented on packages/main/src/plugin/statusbar/statusbar-item.ts:
```diff
@@ -92,6 +93,15 @@ export class StatusBarItemImpl implements StatusBarItem {
     this.update();
   }
 
+  public get notify(): boolean {
```
I would not introduce a notify option that will lead to a lot of blinks everywhere.
We can provide an icon like warning icon

we're following most of the VS Code API and there is no notify so I think we can do without it

benoitf replied:
Thanks

benoitf reviewed (commented):
I think we're missing a menu Entry 'Podman Desktop/Check for Updates...' that performs a check

benoitf commented:
about the test procedure, I'm wondering why we need a minio server ?

For example if I just change the package.json to 0.0.1 and do `yarn compile:current` and then I launch the dmg from the dist folder it'll grab github metadata and see that there is 0.11.0 for example

benoitf commented:
What I would expect is :

1. if there is an update, the tray icon should have a different icon
2. Add an entry in the tray icon menu as well like 'Download update...' or 'Check for Updates...' (I think it's what Docker Desktop does as well)

we've also https://github.com/containers/podman-desktop/issues/1418 so probably we could display the version of Podman Desktop in the status bar and add the warning icon/information icon near the version if we detect that there is an updated version (that would replace the current reload icon from this PR)

deboer-tim commented:
> we could display the version of Podman Desktop in the status bar

Personally I'd rather we don't, it is distracting and irrelevant most of the time. As a user my ideal expectation is only version number in About dialog, a 'Check for updates' that confirms X is the latest version (or not), and the 'an update is available' notification in taskbar.

benoitf commented:
@deboer-tim 
here is the rationale

1. on Linux and Windows you don't have the menu until you click on left ALT key
2. I think it's not like an IDE where you spend all your day in, and most of the users of Podman Desktop are not finding the version.
3. It's also what Docker Desktop does so most of users are expecting to see the version on the same location in Podman Desktop

deboer-tim commented:
I don't use a single other tool (not just IDEs) whose version is always visible/on the status bar - but this is also not a critical issue to me and do not want to hold this up behind other issues. I'm fine putting it there for now and we can always revisit.

benoitf commented:
@deboer-tim we could have an opt-out for that (and make it visible only if there is an update) and add a 'About ...' entry in the tray icon as well.

deboer-tim commented:
@benoitf Let's just keep it simple for now, e.g. no opt-out. We can see how the Dashboard or menus evolve, or feedback we get, and evolve later.

cdrage commented:
> 

It's to test the actual upgrade functionality not notification, see: https://www.electron.build/tutorials/test-update-on-s3-locally and https://github.com/electron-userland/electron-builder/issues/3053

We don't *have* to check to see if the Upgrade will work as it's published to GitHub, but it's nice to have for QE.

cdrage commented:
> about the test procedure, I'm wondering why we need a minio server ?
> 
> For example if I just change the package.json to 0.0.1 and do `yarn compile:current` and then I launch the dmg from the dist folder it'll grab github metadata and see that there is 0.11.0 for example

It's to test the actual upgrade functionality not notification, see: https://www.electron.build/tutorials/test-update-on-s3-locally and https://github.com/electron-userland/electron-builder/issues/3053

We don't *have* to check to see if the Upgrade will work as it's published to GitHub, but it's nice to have for QE.

benoitf commented:
let's use 'always see the version'. If there is an update then we add 'warning' icon in status bar and clicking will do the update.

Then we add in tray menu a 'check for updates..' or 'perform update' ( depending if we already know that there is an update to do or not)

We can defer the tray icon being changed later

jeffmaury commented:
Same as @lstocchi I don't get why we need the Minio server. I have 0.0.2 on Minio and when i run 0.0.1 I tells me 0.11.0 is available. But I can see the flashing icon in the status bar. But I click on it and then choose later, the icon stops to flash

cdrage commented:
> Same as @lstocchi I don't get why we need the Minio server. I have 0.0.2 on Minio and when i run 0.0.1 I tells me 0.11.0 is available. But I can see the flashing icon in the status bar. But I click on it and then choose later, the icon stops to flash

As noted in the doc, it's to test the upgrade functionality. As it won't work unless using minio.

cdrage commented:
@benoitf @lstocchi @jeffmaury Updated and ready for review 💯 

I've also changed it so that it will "spin" if there is an update available as I found it we can actually use `fa-spin` from the fontawesome library.

Let me know if this is okay, otherwise, we can change the icon to something else? Any suggestions @benoitf (I think you suggested maybe the "warning" symbol?

cdrage commented:
For testing, simply change the package.json version to something lower (`0.0.1`), yarn compile and then run the binary.

cdrage commented:
Rebased against main

benoitf commented:
in development mode I see the update

Clone the repo, launch yarn watch

![image](https://user-images.githubusercontent.com/436777/220929495-287ea641-7c57-4f43-ae56-9bafe52310c2.png)

I guess it should not be displayed ?

benoitf commented:
also my current version is `version": "0.13.0-next",`

If I click on the icon (even if I shouldn't have it) I have:

```
Uncaught (in promise) Error: Unknown command: update
await (async)
click_handler @ StatusBarItem.svelte? [sm]:45
```

benoitf commented:
I'm using `4b225bf7295e91acc1dc2aa40a6015dce14af8ab` sha

cdrage commented:
> also my current version is `version": "0.13.0-next",`
> 
> If I click on the icon (even if I shouldn't have it) I have:
> 
> ```
> Uncaught (in promise) Error: Unknown command: update
> await (async)
> click_handler @ StatusBarItem.svelte? [sm]:45
> ```

Thanks for the edge case. I'll cover that and try it out with a version that's higher being ran / testing it that way.

cdrage commented:
![Screenshot 2023-02-23 at 11 12 49 AM](https://user-images.githubusercontent.com/6422176/220975477-6b9debe7-3cf0-4370-9bfd-68166faeb3c3.png)
![Screenshot 2023-02-23 at 11 51 32 AM](https://user-images.githubusercontent.com/6422176/220976509-1671b1c8-b4a4-4814-9913-1b4198ce5df4.png)


Ready for review!

* Button will only show in production / compiled version
* Shows dialog what version you are on when you click
* If version available, the icon is the "refresh" symbol and spins. Clicking it shows update dialog. Clicking later / update stops spin.

How to test notification:
1. Set your `package.json` version to something lower `0.0.1` (not `0.0.1-next`)
2. You'll now see the spinner refresh icon

@benoitf 

For testing the "update" functionality:
1. You'll have to merge this PR and then create a "release" of something like `0.0.1` to GitHub. 
2. Download the file, copy over to `Applications` in macOS.
3. It should show that a new update is available.
4. Clicking "Update" in the dialog should download the update / replace Podman Desktop.

benoitf commented:
it's a matter of taste but I would expect that if I'm on the latest version, I don't need to click on the version. Tooltip already notify me that I'm on the latest version. And then I shouldn't get anything if I click on the icon.

next, IMHO the icon does not fit well to say that there is something new
I would expect more the https://fontawesome.com/icons/exclamation-triangle?s=solid&f=classic or  https://fontawesome.com/icons/circle-info?s=light&f=classic or https://fontawesome.com/icons/download?s=solid&f=classic but I guess @mairin is the lead there

The icon should be a prefix of the title. So if I'm on the `v0.1` I would expect to see `<icon> v0.1` and not only `<icon>` if there is action possible (like to download an update) but I should still the version I'm using.

cdrage commented:
> it's a matter of taste but I would expect that if I'm on the latest version, I don't need to click on the version. Tooltip already notify me that I'm on the latest version. And then I shouldn't get anything if I click on the icon.
> 
> next, IMHO the icon does not fit well to say that there is something new I would expect more the https://fontawesome.com/icons/exclamation-triangle?s=solid&f=classic or https://fontawesome.com/icons/circle-info?s=light&f=classic or https://fontawesome.com/icons/download?s=solid&f=classic but I guess @mairin is the lead there
> 
> The icon should be a prefix of the title. So if I'm on the `v0.1` I would expect to see `<icon> v0.1` and not only `<icon>` if there is action possible (like to download an update) but I should still the version I'm using.

I think we're finally getting there with the UI! I tried the icon + version number but even with the "download" icon, it wasn't very present. I propose we also added (Update Available) to the text.

I've updated it to the following:
* No click / dialog when clicking on version
* Updated icon

When update is NOT available it'll show version:
![Screenshot 2023-02-23 at 1 47 31 PM](https://user-images.githubusercontent.com/6422176/221002780-0d977a2a-6f75-48ca-b72c-0746ac128286.png)

When update IS available, it'll show the following:
![Screenshot 2023-02-23 at 1 50 26 PM](https://user-images.githubusercontent.com/6422176/221002792-1a5f9a14-e9a9-4e4c-8466-e8366937e544.png)

benoitf commented:
I think the text ( update available) is not necessary and takes too much space.
Maybe let's give a try with the information or warning icon instead. It should bring attention to the user.

cdrage commented:
> I think the text ( update available) is not necessary and takes too much space. Maybe let's give a try with the information or warning icon instead. It should bring attention to the user.

Updated! Removed the (update available) text and now it shows the triangle exclamation beside the version name.

benoitf commented on packages/main/src/plugin/index.ts:
```diff
@@ -306,6 +309,86 @@ export class PluginSystem {
+
+        // autoUpdater.checkForUpdatesAndNotify() is called from main/src/index.ts
+        autoUpdater.on('update-available', () => {
+          // If there is an update available, remove the "version" entry from the status bar
+          // and add the "update" entry
+          statusBarRegistry.removeEntry('version');
+
+          // Will add the new entry.
+          // If there is already an entry available, it'll update the icon to the new one (if it's different)
+          statusBarRegistry.setEntry(
+            'update',
+            false,
+            0,
+            currentVersion,
+            'Update available',
+            UPDATER_UPDATE_AVAILABLE_ICON,
+            true,
+            'update',
+            undefined,
+          );
+        });
```
I'm wondering why we don't have a single entry that do all the stuff ?
why add a new item and remove the previous one. we can just handle only one item in the status bar and depending on the context, update the icon/text, etc ?

cdrage replied:
> I'm wondering why we don't have a single entry that do all the stuff ? why add a new item and remove the previous one. we can just handle only one item in the status bar and depending on the context, update the icon/text, etc ?

Instead of removing / adding, would it make more sense if we just set enabled / disabled? We could do that instead. Setup both items at the beginning and then enable / disable in the logic.

benoitf commented on packages/main/src/plugin/index.ts:
```diff
@@ -306,6 +309,86 @@ export class PluginSystem {
+          // for us to retrieve the version used
+          autoUpdater.checkForUpdates().then(update => {
+            // Get the version if autoUpdater was able to retrieve update (not undefined)
+            const updateVersion = update?.updateInfo.version ? `(v${update?.updateInfo.version}) ` : '';
```
```suggestion
            const updateVersion = update?.updateInfo.version ? `(v${update?.updateInfo.version})` : '';
```

benoitf replied:
why is it within `()` ?

cdrage replied:
intentional in case we're unable to get updateVersion from `update?.updateInfo.version`. Although I doubt that'll happen.

benoitf replied:
but it will display for example `A new version (v1.1)` vs `A new version v1.1`

cdrage replied:
Ah yeah, I had that so it would match what some other Electron apps do. I'll update it to show `A new version v1.1` instead of the brackets.

benoitf commented on packages/main/src/plugin/index.ts:
```diff
@@ -306,6 +309,86 @@ export class PluginSystem {
+              .showMessageBox({
+                type: 'info',
+                title: 'Update Available',
+                message: `A new version ${updateVersion}of Podman Desktop is available. Do you want to update your current version (${currentVersion})?`,
```
```suggestion
                message: `A new version ${updateVersion} of Podman Desktop is available. Do you want to update your current version (${currentVersion})?`,
```

cdrage replied:
the space is intentional in case there is an issue getting updateVersion (see above other code)

benoitf commented on packages/main/src/plugin/statusbar/statusbar-registry.ts:
```diff
@@ -51,6 +51,15 @@ export class StatusBarRegistry implements IDisposable {
     }
   }
 
+  // Change the icon of an entry and send an event to the renderer
+  changeEntryIcon(id: string, iconClass: string) {
+    const entry = this.entries.get(id);
+    if (entry) {
+      entry.entry.activeIconClass = iconClass;
+      this.apiSender.send(STATUS_BAR_UPDATED_EVENT_NAME, undefined);
+    }
+  }
+
```
AFAIK we should not need that

entry belongs to someone, we can't change something that we don't own

cdrage commented:
@benoitf Updated with your suggestions (no longer removing + readding an entry, just updating the current one).


MAJOR NOTE: This has `0.0.1` committed within `package.json` so we can check it against the GitHub-built binaries (look at GitHub actions).

Must remove this before merge..

benoitf reviewed (commented):
status bar entry is there and warns me that there is an update
but as it's unsigned (local build) there is an error but it's not handled by the UI
so I'm waiting that the update is fetched but I don't see any error in the UI (only in the dev console)


https://user-images.githubusercontent.com/436777/222078747-7f75ba51-2297-4a56-babb-696f1cbaec4c.mp4


- In development mode I would expect to still see the version of the product in the status bar (but we don't check if there is a new version). But with the PR there is no status bar element.

cdrage commented:
> 

Need some clarification for the below parts:

*For the error:*

That's one of the issues that we had discussed earlier and why I had to write the documentation (CONTRIBUTING.md) for running it against a minio server. Those errors are normal even with the `main` branch and checking for updates. `yarn compile` doesn't sign the binaries unless you're doing a production push. The Update button won't work (even on main) unless it's either a signed binary / was pushed to GitHub, OR you follow the `CONTRIBUTING.md` doc for running it against a minio server.


*For the UI:*

We discussed that the version not appear when running `yarn watch` / dev mode earlier? Should we re-add it but not have the update button?


*Further Tasks:*
* Should I try / catch the error?
* Should I re-add showing the version in `yarn watch`?

benoitf commented:
> 
>For the error:
>
>That's one of the issues that we had discussed earlier and why I had to write the documentation (CONTRIBUTING.md) for >running it against a minio server. Those errors are normal even with the main branch and checking for updates. yarn >compile doesn't sign the binaries unless you're doing a production push. The Update button won't work (even on main) >unless it's either a signed binary / was pushed to GitHub, OR you follow the CONTRIBUTING.md doc for running it against a > minio server.

> Even in the main branch, we don't try / catch / show the error within the UI. Should we add it with this PR?

I know the reason of the error, just that user needs to be informed in the case of the error.
For the main branch, there is no explicit action of the user (it's done under the hood) so it's ok if error is not reported.
But here, user explicitly click on a button to download the update and nothing happens. So the user don't know that something failed or something succeed. 

>
>For the UI:

> We discussed that the version not appear when running yarn watch / dev mode earlier? Should we re-add it but not have the update button?

AFAIK we discussed that we should not check for updates in development mode, but not that the version should not be displayed in the status bar.

cdrage commented:
> 

Thanks for the clarification! I'll update this PR and ping you when I've made the changes.

cdrage commented:
<img width="630" alt="Screenshot 2023-03-02 at 11 17 29 AM" src="https://user-images.githubusercontent.com/6422176/222491546-4e460dac-e026-4830-8c0b-e9379d282c63.png">
<img width="1050" alt="Screenshot 2023-03-02 at 11 09 45 AM" src="https://user-images.githubusercontent.com/6422176/222491551-65c724ef-8298-45b4-9580-a4db2e77e584.png">
<img width="1132" alt="Screenshot 2023-03-02 at 11 04 06 AM" src="https://user-images.githubusercontent.com/6422176/222491555-245fe3a2-ac11-4490-bc4b-83dd730ae339.png">


Updated PR and ready for another review:

* Added "error" message when trying to press "Update" when using `yarn compile` to test
* Version now shows in dev mode 
* The "Your version is up to date" message only shows in PROD and if you are not using Linux (since electro autoupdater only works on macOS and Windows).

As a kind reminder, if we merge this, we must make a github production release (label it 0.0.1 and do not publish it anywhere else, and then promptly remove it after) in order to properly test the functionality in production before we release a larger release.

benoitf reviewed (approved):
code is ok, version is displayed (in dev mode as well)
I noticed that in case of error we say to look at the devconsole. We can't have the error in the dialog box ?

I notarized/signed locally a build but I was not able to update it  🤔  (error is different than the unsigned problem)



https://user-images.githubusercontent.com/436777/222767294-ca64aa2a-3771-40f6-bca1-95e52ac37d17.mp4

cdrage commented:
Yeah! I encountered a similar issue except for when I tested it against the minio server. 

We'll have to merge this in (after the f2f) and do a test release after to test the functionality.

cdrage commented:
@benoitf We could do that. I think it would be safer however if we could instead release 0.13.0 *first* and then merge this after. We can then check with the next channel (test release a 0.13.1) to see if it builds correctly. WDYT

benoitf commented:
we can merge it now, it should check against the next build

I mean, latest build check for latest build (like 0.13.0 will check for 0.13.1)
but for example [v0.0.202303220902-9e36beb](https://github.com/containers/podman-desktop/releases/tag/v0.0.202303220902-9e36beb) will check for a new one as well
